rskj-core/src/main/java/co/rsk/net/light/LightSyncProcessor.java

```diff
@@ -0,0 +1,190 @@
+                        protocolVersion);
+                lightPeer.disconnect(ReasonCode.INCOMPATIBLE_PROTOCOL);
+                ctx.pipeline().remove(lightClientHandler); // Peer is not compatible for the 'lc' sub-protocol
+                return;
+            }
+
+            int networkId = config.networkId();
+            int msgNetworkId = status.getNetworkId();
+            if (msgNetworkId != networkId) {
+                loggerNet.info("Removing LCHandler for {} due to invalid network", ctx.channel().remoteAddress());
+                loggerNet.info("Different network received: config network ID {} - message network ID {}",
+                        networkId, msgNetworkId);
+                lightPeer.disconnect(ReasonCode.NULL_IDENTITY);
+                ctx.pipeline().remove(lightClientHandler);
+                return;
+            }
+
+            Keccak256 genesisHash = genesis.getHash();
+            Keccak256 msgGenesisHash = new Keccak256(status.getGenesisHash());
+            if (!msgGenesisHash.equals(genesisHash)) {
+                loggerNet.info("Removing LCHandler for {} due to unexpected genesis", ctx.channel().remoteAddress());
+                loggerNet.info("Config genesis hash {} - message genesis hash {}",
+                        genesisHash, msgGenesisHash);
+                lightPeer.disconnect(ReasonCode.UNEXPECTED_GENESIS);
+                ctx.pipeline().remove(lightClientHandler);
+                return;
+            }
+        } catch (NoSuchElementException e) {
+            loggerNet.debug("LCHandler already removed - exception: {}", e.getMessage());
+        }
+
+        if (peerStatuses.size() >= MAX_PEER_CONNECTIONS) {
+            return;
+        }
+
+        if (!hasLowerDifficulty(status)) {
+            return;
+        }
+
+        peerStatuses.put(lightPeer, status);
+
+        if (msg.isTxRelay()) {
+            txRelay.put(lightPeer, true);
+        }
+
+        byte[] bestBlockHash = status.getBestHash();
+        GetBlockHeaderMessage blockHeaderMessage = new GetBlockHeaderMessage(++lastRequestedId, bestBlockHash);
+        pendingMessages.put(lastRequestedId, BLOCK_HEADER);
+        lightPeer.sendMessage(blockHeaderMessage);
+    }
+
+    public void sendStatusMessage(LightPeer lightPeer) {
+        Block block = blockStore.getBestBlock();
+        LightStatus status = getCurrentStatus(block);
+        StatusMessage statusMessage = new StatusMessage(0L, status, false);
+
+        lightPeer.sendMessage(statusMessage);
+
+        loggerNet.trace("Sending status best block {} to {}",
+                block.getNumber(), lightPeer.getPeerIdShort());
+    }
+
+    public void processBlockHeaderMessage(long id, BlockHeader blockHeader, LightPeer lightPeer) {
+        if (!isPending(id, BLOCK_HEADER)) {
+            return;
+        }
+
+        pendingMessages.remove(id, BLOCK_HEADER);
+        lightPeer.receivedBlock(blockHeader);
+    }
+
+    public boolean hasTxRelay(LightPeer peer) {
```

Comment:
hasTxRelay should have a @VisibleForTesting annotation, or similar. Because it is a method ONLY for testing

Reply:
At this moment, it should be visible for testing, right.